pyatv/interface.py

```diff
@@ -410,6 +410,43 @@ def connection_closed(self):
         """Device connection was (intentionally) closed."""
         raise NotImplementedError()
 
+class PowerListener:
+    """Listener interface for power updates."""
+
+    @abstractmethod
+    def powerstate_update(self, old_state, new_state):
+        """Device power state was updated."""
+        raise NotImplementedError()
+
+class Power:
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        """Initialize a new AppleTV."""
+        self.__listener = None
+
+    @property
+    def listener(self):
+        """Object receiving power state updates.
+
+        Must be an object conforming to PowerListener.
+        """
+        return self.__listener
+
+    @abstractproperty
+    def power_state(self):
+        """Return device power state."""
+        raise exceptions.NotSupportedError()
+
+    @abstractmethod
+    def turn_on(self):
```

Comment:
Ah, this was bad on my end. This method should be async:
```suggestion
    async def turn_on(self):
```